packages/neon/neon_talk/lib/src/blocs/room.dart

```diff
@@ -43,6 +43,38 @@ class _TalkRoomBloc extends InteractiveBloc implements TalkRoomBloc {
     token = room.token;
     this.room.add(Result.success(room));
 
+    messages.listen((result) {
+      if (!result.hasSuccessfulData) {
+        return;
+      }
+
+      // TODO: More room fields might be changed based on the received messages (e.g. room was renamed).
+
+      final value = this.room.value;
+      final lastMessage = result.requireData.first;
```

Comment:
You're right, there could be no messages. When the room is created there will always be a system message, but the automatically expiring messages could be enabled. I think I assumed this because when sending a new message or polling new messages there is at least one for sure, but for the initial loading of the messages this might not be true ofc.